### PR TITLE
WIP: document new alertmanager_storage config

### PR DESCRIPTION
**Pull request by ShuzZzle**
**What this PR does**:
Update documentation on how to configure alertmanager_storage

**Which issue(s) this PR fixes**:
#4206 

**Checklist**
- [ / ] Tests updated
- [ X ] Documentation added
- [ / ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/guides/alert-manager-configuration.md [hunk 1]
```diff
@@ -9,6 +9,23 @@ slug: alertmanager-configuration
...
+```YAML
+alertmanager_storage:
+  backend: "s3"
+  s3:
+    endpoint: <endpoint>
+    bucket_name: <bucket>
+    secret_access_key: <secret_access_key>
+    access_key_id: <access_key_id>
+```
+All blocks storage bucket clients and the config service are currently supported. If you choose to try the experimental Alertmanager resharding feature then please note that the local and configdb storage backends are not supported.
```
Comment (stevesg) [inline docs/guides/alert-manager-configuration.md:24]: > All blocks storage bucket clients and the config service are currently supported. 

I think this is redundant - I would re-work this to include a reference the auto-generated documentation; this is kept in sync with the code: https://cortexmetrics.io/docs/configuration/configuration-file/#alertmanager_storage_config

> If you choose to try the experimental Alertmanager resharding feature then please note that the local and configdb storage backends are not supported.

I would be more explicit here, for example:

> Note that when using `-alertmanager.sharding-enabled=true`, the following storage backends are not supported: `local`, `configdb`.

**Review by stevesg — COMMENTED** [review]
Thanks for improving this documentation. Saw this was draft but I added some pointers for you anyway. I'll be happy to re-review once you're finished.

**Comment by stevesg** [comment]
Also, don't forget to sign your commits (DCO). You'll probably be best off squashing your commits into one, and then you can add the sign off to your previous commit like this:

```
git commit -s --amend --no-edit
```

File: docs/guides/alert-manager-configuration.md (hunk 1 above)
Comment (ShuzZzle) [reply docs/guides/alert-manager-configuration.md]: Thank you so much. As you can probably tell english is not my first english but hey :D one can try

I agree with the redundant section and reworked it a bit and linked back to the configuration file. If you don't have anything more to add then I would like to rebase and squash my commits.

**Comment by ShuzZzle** [comment]
I am gonna redo this :P

**Pull request closed without merging**